### PR TITLE
First version of a contributing file

### PR DESCRIPTION
**Pull request by pamasse**
First version of a contributing file #25. It quickly explains how the code is structured and details how to install the development environment.

### CODE REVIEW

**Comment by sandoche2k** [comment]
@pamasse close?

File: README.md [hunk 1]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+|   +-- romanize.pipe.*             : Transform latin number to roman number (1 -> I, 2 -> II)
+|   +-- safe-html.pipe.*            : Sanitize text to HTML
++-- services
+|   +-- alert.service.*     : to display alert messages
+|   +-- app.service.*       : to manage app configuration (api url, etc.)
+|   +-- dns-check.service.* : to communicate with the Zonemaster Backend API 
++-- app.*
+```
+
+The ``components`` folder is composed by subfolder that represent main components.
```
Comment (matsduf) [inline README.md]: Suggest:

```
The ``components`` folder is composed of subfolders that represent the main components.
```

File: README.md [hunk 2]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+Each folder of component contain four files :
```
Comment (matsduf) [inline README.md]: But you only list three.

File: README.md [hunk 3]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+  - ``*.component.css``   : The component style 
+  - ``*.component.html``   : The component html 
+  - ``*.component.*.ts``   : The component typescript
```
Comment (matsduf) [inline README.md]: Suggest:
```
- ``*.component.ts``   : The component typescript 
- ``*.component.spec.ts``   : The component typescript 
```

File: README.md [hunk 4]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+```
+
+The ``components`` folder is composed by subfolder that represent main components.
+Each folder of component contain four files :
+  - ``*.component.css``   : The component style 
+  - ``*.component.html``   : The component html 
+  - ``*.component.*.ts``   : The component typescript 
+    
+All assets are available in ``src/assets`` folder. 
+It's split by concern, all the javascript is in the ``js`` folder, the style in the ``css``, etc.
```
Comment (matsduf) [inline README.md]: There is a ccs directory, but no js directory

File: README.md [hunk 5]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+The configurations files of the application are in the ``src/environments`` folder.
+``environment.ts`` is use for development purpose, and the ``environment.prod.ts`` for production.
+
+    
 In order to contribute
 * Install [Nodejs](https://nodejs.org) 
-* Fork then Clone your repo 
-* Go to the folder and install project dependencies `npm install`
-* Update environment files in `/src/environments/` to add a backend api endpoint
+* Fork and Clone the repo
```
Comment (matsduf) [inline README.md]: Suggest:
```
* Fork the Zonemaster GUI repository on Github into your own user on Github.
* Clone your fork to your working environment.

File: README.md [hunk 6]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
+    
 In order to contribute
 * Install [Nodejs](https://nodejs.org) 
-* Fork then Clone your repo 
-* Go to the folder and install project dependencies `npm install`
-* Update environment files in `/src/environments/` to add a backend api endpoint
+* Fork and Clone the repo 
+* Go to the folder and install project dependencies with `npm install`
+* Update environment files in `/src/environments/` to add a backend api endpoint (or leave the default)
+* Make your change, test it and create a Pull Request
```
Comment (matsduf) [inline README.md]: * Make your changes, test them and push them to your fork on Github
* From your fork, make a Pull Request against the zonemaster/zonemaster-gui repository. Please always
  make the Pull Request against the develop branch.
* Thank you for your controbution!

File: README.md [hunk 7]
```diff
@@ -4,31 +4,70 @@ Zonemaster Web GUI
...
 
-You are now ready to develop 
+You are now ready to contribute !
```
Comment (matsduf) [inline README.md]: If you want this statement, I suggest that you move it before "Make your changes". Remove the extra space between "contribute" and "!".

File: README.md [hunk 8]
```diff
@@ -4,31 +4,73 @@ Zonemaster Web GUI
...
+|   +-- app.service.*       : to manage app configuration (api url, etc.)
+|   +-- dns-check.service.* : to communicate with the Zonemaster Backend API 
++-- app.*
+```
+
+The ``components`` folder is composed of subfolders that represent the main components.
+Each folder of component contain three files :
+  - ``*.component.css``       : The component's style 
+  - ``*.component.html``      : The component's html 
+  - ``*.component.*.ts``      : The component's typescript
```
Comment (matsduf) [inline README.md]: As I can see it you have remove the *.component.spec.ts files. Then I guess the text should be

```
  - ``*.component.ts``      : The component's typescript 
```

**Comment by sandoche2k** [comment]
@pamasse @matsduf is waiting for your response.

**Pull request merged**